File: Estágio AutoMind/Usuario/Menu.cs
namespace EstagioAutomind.Usuario;

public class Menu {
	public static int menu() {
		int opcaoMenu = -1;
		bool menuValidador = false; /* booleano para validação */
		
		while (!menuValidador) {
			Console.WriteLine("\n*** Menu ***");
			Console.WriteLine("0 - sair");
			Console.WriteLine("1 - Cadastrar usuário");
			Console.WriteLine("2 - Listar usuários");
			Console.WriteLine("3 - Buscar usuário");
			Console.Write("Insira alguma opção para continuar: ");
			string entrada = Console.ReadLine();
				
			/* verifica se a entrada pode ser convertida para inteiro */
			if (int.TryParse(entrada, out opcaoMenu)) {
				
				/* se for um inteiro, validar a opção */
				if (opcaoMenu >= 0 && opcaoMenu <= 3) {
					menuValidador = true;
					opcaoMenu = int.Parse(entrada);
				} else {  Console.WriteLine("Opção inválida. Por favor, insira um número entre 0 e 3."); }
				
			} else { Console.WriteLine("Entrada inválida. Por favor, digite um número válido."); }
		}
		return opcaoMenu;
	}
}

File: Estágio AutoMind [Java]/Automind_java/src/Usuario/Menu.java
package Usuario;

import java.util.Scanner;

public class Menu {
    public static int menu(Scanner sc){

        int opcao = -1, temp;
        boolean validador = false;

        while (!validador) {
            System.out.println("*** Menu ***");
            System.out.println("0. Sair");
            System.out.println("1. Cadastrar usuário");
            System.out.println("2. Listar usuário");
            System.out.println("3. Buscar usuário");
            System.out.print("Insira alguma opção para continuar: ");
            String entrada = sc.nextLine();

            try {
                /* Verificar se a entrada recebida foi um inteiro */
                temp = Integer.parseInt(entrada);
                /* Verificar se o número está dentro do intervalo permitido */
                if (temp >= 0 && temp <= 3) {
                    opcao = temp;
                    validador = true;
                } else {
                    System.out.println("ERRO! Insira um valor entre 0 e 3!");
                }
            } catch (NumberFormatException e) {
                System.out.println("Erro: Entrada inválida, só é permitido número inteiro");
            }
        }

        return opcao;
    }

}
